packages/backend/src/core/PerpetualValidiumSyncService.ts

```diff
@@ -42,23 +50,34 @@ export class PerpetualValidiumSyncService implements IDataSyncService {
       userRegistrations: userRegistrations.length,
     })
 
-    for (const transition of stateTransitions) {
+    const stateUpdates = await this.processStateUpdates(stateTransitions)
+    await this.l2TransactionDownloader?.sync(stateUpdates)
```

Comment:
I don't think we need to get a list of stateUpdates processed here. We just want the newest stateUpdateId and then the `l2TransactionDownloader` should sync up to that.

Reply:
If we can get it from processStateUpdates then why would query database later for batchId etc.?